proof/crefine/X64/Interrupt_C.thy

```diff
@@ -530,32 +530,61 @@ lemma ccorres_pre_gets_x64KSNumIOAPICs_ksArchState:
   apply clarsimp
   done
 
+lemma ccorres_pre_gets_x64KSIOAPICnIRQs_ksArchState:
+  assumes cc: "\<And>rv. ccorres r xf (P rv) (P' rv) hs (f rv) c"
+  shows   "ccorres r xf
+                   (\<lambda>s. (\<forall>rv. x64KSIOAPICnIRQs (ksArchState s) = rv  \<longrightarrow> P rv s))
+                   {s. \<forall>rv. s \<in> P' rv } hs
+                   (gets (x64KSIOAPICnIRQs \<circ> ksArchState) >>= (\<lambda>rv. f rv)) c"
+  apply (rule ccorres_guard_imp)
+    apply (rule ccorres_symb_exec_l)
+       defer
```

Comment:
this is a bit of a hot-take, but looks like the other proofs, so I guess we'll keep it

Reply:
It is a copy-paste of other `*_pre` lemmas, yes. I tried fiddling with it for a bit, but didn't come up with something nicer in reasonable time.